In [2]:
!pip install evaluate

  Using cached xxhash-3.5.0-cp312-cp312-win_amd64.whl.metadata (13 kB)
  Using cached dill-0.3.8-py3-none-any.whl.metadata (10 kB)
  Using cached multiprocess-0.70.16-py312-none-any.whl.metadata (7.2 kB)
   ---------------------------------------- 0.0/84.1 kB ? eta -:--:--
   ---------------------------------------- 84.1/84.1 kB 4.6 MB/s eta 0:00:00
   ---------------------------------------- 0.0/494.8 kB ? eta -:--:--
   --------------------------------------  491.5/494.8 kB 10.2 MB/s eta 0:00:01
   ---------------------------------------- 494.8/494.8 kB 7.8 MB/s eta 0:00:00
Using cached dill-0.3.8-py3-none-any.whl (116 kB)
   ---------------------------------------- 0.0/193.6 kB ? eta -:--:--
   ---------------------------------------- 0.0/193.6 kB ? eta -:--:--
   -- ------------------------------------- 10.2/193.6 kB ? eta -:--:--
   -------------------------------------- - 184.3/193.6 kB 3.7 MB/s eta 0:00:01
   ---------------------------------------- 193.6/193.6 kB 3.9 MB/s eta 0

In [8]:
predictions = ["""From Joshi et al 2003, increasing calcium content by preacidifying milk brought several desirable changes in functional properties of part skim Mozzarella cheese. These changes included increased melt area, flow rate, and extent of flow of cheese. 
Details included a desireable reduction of approximately 35% calcium that almost doubled the melt area of the cheese. 
Lowering of calcium also caused reduction in time-temperature required for softening and melting of cheese."""]

In [4]:
references = ["""From Joshi et al 2003, lowering calcium content by preacidifying milk brought several desirable changes in functional properties of part skim Mozzarella cheese. These changes included increased melt area, flow rate, and extent of flow of cheese. 
Details included a desireable reduction of approximately 35% calcium that almost doubled the melt area of the cheese. 
Lowering of calcium also caused reduction in time-temperature required for softening and melting of cheese."""]

In [9]:
from evaluate import load
bertscore = load("bertscore")
results = bertscore.compute(predictions=predictions, references=references, lang="en")
print(results)

Some weights of RobertaModel were not initialized from the model checkpoint at roberta-large and are newly initialized: ['pooler.dense.bias', 'pooler.dense.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


{'precision': [0.9975027441978455], 'recall': [0.9975027441978455], 'f1': [0.9975027441978455], 'hashcode': 'roberta-large_L17_no-idf_version=0.3.12(hug_trans=4.53.1)'}


In the two examples above you can see that references reads "lowering calcium content..." but prediction reads "increasing calcium content". The prediction gives completely wrong answer which is not captured by BERTScore which return precision of 0.9975. Therefore, BERTScore is not able to capture crucial differences in meaning and was rejected by me as a metric for accuracy. It is still a good metric to compare similarity of wording between two texts.

In applications where correct answers are critical, BERTScore cannot be used as a sole metric.